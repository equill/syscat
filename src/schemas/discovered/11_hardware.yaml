name: hardware_discovered
version: 1

resourcetypes:
  networkInterfaces:
    attributes:
      description:
      macaddress:
      snmpindex:
      ifname:
      ifdescr:
      ifalias:
      iftype:
      ifspeed:
      ifhighspeed:
      ifphysaddress:
    dependent: 'true'
