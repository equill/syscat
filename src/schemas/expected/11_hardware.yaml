name: hardware_expected
version: 1

resourcetypes:
  deviceRoles:
    attributes:
      description:
    notes: Classification of the device, e.g. host, router or switch.
  environments:
    attributes:
      description:
    notes: Development, testing, staging, production etc.
  lifeCyclePhases:
    attributes:
      description:
    notes: Where the device currently is, in its lifecycle. E.g. ordered, installed, in service or decommissioned.
  networkInterfaces:
    attributes:
      description:
      monitor:
        description: Whether this interface is to be monitored.
    dependent: 'true'
  serviceLevels:
    attributes:
      description:
    notes: Sets the expectations for responses to issues, e.g. gold, silver, bronze, plastic

relationships:
- uri: /models/Capability/deviceRoles
  cardinality: many:many
  notes: "The capabilities available from this model, regardless of whether you're using any/all of them right now."
- uri: /devices/Role/deviceRoles
  cardinality: many:many
  notes: "The role(s) being fulfulled by this device: e.g, a layer-3 switch should have the roles switch and router."
- uri: /devices/LifeCyclePhase/lifeCyclePhases
  cardinality: many:1
- uri: /devices/Environment/environments
  cardinality: many:1
- uri: /devices/ServiceLevel/serviceLevels
  cardinality: many:1
- uri: /devices/Vendor/organisations
  cardinality: many:1
  notes: The organisation from which this device was bought.
- uri: /parts/Vendor/organisations
  cardinality: many:1
- uri: /makes/Produces/parts
  cardinality: 1:many
  dependent: 'true'
- uri: /devices/TechnicalOwner/people
  cardinality: many:1
- uri: /devices/BusinessOwner/people
  cardinality: many:1
- uri: /devices/BusinessOwner/organisations
  cardinality: many:1
- uri: /contracts/Supplies/devices
  cardinality: 1:many
- uri: /contracts/Supports/devices
  cardinality: 1:many
- uri: /contracts/Supplies/parts
  cardinality: 1:many
- uri: /contracts/Supports/parts
  cardinality: 1:many
